index.js

```diff
@@ -59,47 +91,23 @@ function router (controller, routes, options) {
       delete input.params;
       delete input.query;
 
-      var params = route.match(/:.[^\/]*/g);
-      var url = route;
-
-      if (params) {
-        // If called from a url change, add params and query to input
-        if (input.route.params) {
-          input = params.reduce(function (input, param) {
-            var key = param.substr(1, param.length);
-            input[key] = input.route.params[key];
-            return input;
-          }, input);
-        }
+      var url = getUrl(route, input);
 
-        // Create url based on direct signal input or
-        // params passed from addressbar
-        url = params.reduce(function (url, param) {
-          var key = param.substr(1, param.length);
-          if (!(key in input)) {
-            throw new Error('Cerebral router - The signal "' + routes[route] + '" is bound to "' + route + '" route, but required param "' + key + '" wasn\'t provided.');
-          }
-          return url.replace(param, input[key] || '');
-        }, url);
-
-        // Check resulted url still matches given route
-        var urlMatched = false;
-        var checkRoute = {};
+      // Check resulted url still matches given route
+      var urlMatched = false;
+      var checkRoute = {};
 
-        checkRoute[route] = function () {
-          urlMatched = true;
-        };
+      checkRoute[route] = function () {
+        urlMatched = true;
+      };
 
-        urlMapper(url, checkRoute);
+      urlMapper(url, checkRoute);
 
-        if (!urlMatched) {
-          throw new Error('Cerebral router - Computed url for signal "' + routes[route] +'" can\'t match given route "' + route + '".\n' +
-                          'Check required params provided to signal is not falsy.');
-        }
+      if (!urlMatched) {
+        throw new Error('Cerebral router - Computed url for signal "' + routes[route] +'" can\'t match given route "' + route + '".\n' +
+                        'Check required params provided to signal is not falsy.');
```

Comment:
Yes, should be something like the above? "... there are missing keys in the params or no params passed". As:

```js
{
  show: false
}
```
Is a valid params to url: /popup/show/false (/popup/show/:state). Bad example, but yeah, it is supported :-)